### PR TITLE
Various test and functionality fixes for test262

### PR DESCRIPTION
**Pull request by jackhorton**
* added string normalization back to Intl-ICU's localeCompare
* re-organized Collator tests to use UnitTestFramework and live all in one file
* update platform.formatDateTime to use TimeClip
* add TimeClip tests for Intl
* fix time zone canonicalization for default time zones

### CODE REVIEW

**Comment by jackhorton** [comment]
As of this PR, ICU passes 418/588 at the tip of test262 in the Intl402 folder, while WinGlob passes 372/588, with only one test passing for WG but not ICU (and, in ICU, we match V8's behavior). Additionally, ICU takes ~14.5 seconds compared to 18.5 seconds for WinGlob on my dev box.

File: lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp [hunk 1]
```diff
@@ -1843,17 +1807,82 @@ namespace Js
...
+        {
+            rightNormalized = right->GetSz();
+            rightNormalizedLength = right->GetLength();
+        }
+        else
+        {
+            rightNormalized = right->GetNormalizedString(UnicodeText::NormalizationForm::C, tempAllocator, rightNormalizedLength);
+        }
+
+        END_TEMP_ALLOCATOR(tempAllocator, scriptContext);
```
Comment (sethbrenith) [inline lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp]: This smells like a use-after-free, unless I'm misinterpreting the meaning of END_TEMP_ALLOCATOR. Should we do this after the part where we use leftNormalized and rightNormalized below?

**Comment by jackhorton** [comment]
@sethbrenith look better now? Also, any thoughts on where TimeClip should go?

**Comment by sethbrenith** [comment]
DateUtilities sounds promising, but I don't feel strongly about where it goes.

---
In reply to: [379858432](https://github.com/Microsoft/ChakraCore/pull/4949#issuecomment-379858432) [](ancestors = 379858432)

**Comment by jackhorton** [comment]
@dotnet-bot test OSX static_osx_osx_debug please

**Comment by dilijev** [comment]
FWIW LGTM

**Pull request merged**